flink-runtime/src/main/java/org/apache/flink/runtime/util/ZooKeeperUtils.java

```diff
@@ -279,12 +280,13 @@ public static CompletedCheckpointStore createCompletedCheckpoints(
 
 		checkpointsPath += ZooKeeperSubmittedJobGraphStore.getPathForJob(jobId);
 
-		return new ZooKeeperCompletedCheckpointStore(
+		final ZooKeeperCompletedCheckpointStore zooKeeperCompletedCheckpointStore = new ZooKeeperCompletedCheckpointStore(
 			maxNumberOfCheckpointsToRetain,
-			client,
-			checkpointsPath,
-			stateStorage,
+			new ZooKeeperStateHandleStore<>(useNamespaceAndEnsurePath(client, checkpointsPath), stateStorage),
```

Comment:
This could be factored out into its own method `createZooKeeperStateHandlerStore(CuratorFramework, String, RetrievableStateStorage)`. Then we could reuse this call in the test cases which need to instantiate a `ZooKeeperStateHandleStore`.

Reply:
fixed, do you want to review again?